instrumentation/gruf/lib/opentelemetry/instrumentation/gruf/interceptors/server.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/trace/status'
+
+module OpenTelemetry
+  module Instrumentation
+    module Gruf
+      module Interceptors
+        class Server < ::Gruf::Interceptors::ServerInterceptor
+          # rubocop:disable Metrics/MethodLength
+          def call
+            method = request.method_name
+
+            if (instrumentation_config[:grpc_ignore_methods] || []).include?(method)
+              OpenTelemetry::Common::Utilities.untraced do
+                return yield
+              end
+            end
+
+            service_name = request.service.service_name.to_s
+            method_name = request.method_key.to_s
+            route = "/#{service_name}/#{method_name.camelize}"
+
+            attributes = {
+              'component' => 'gRPC',
+              'span.kind' => 'server',
+              'grpc.method_type' => 'request_response',
+              'grpc.service_name' => service_name,
+              'grpc.method_name' => method_name
+            }
+            # rubocop:disable Style/IfUnlessModifier
+            if instrumentation_config[:peer_service]
+              attributes['peer.service'] = instrumentation_config[:peer_service]
+            end
+            # rubocop:enable Style/IfUnlessModifier
+
+            extracted_context = OpenTelemetry.propagation.extract(request.active_call.metadata)
+            OpenTelemetry::Context.with_current(extracted_context) do
+              in_span(
+                create_span_name(route, request, attributes['peer.service']),
+                attributes: attributes
+              ) do |request_span|
+                if instrumentation_config[:log_requests_on_server]
+                  request_span.add_event(
+                    'request',
+                    attributes: {
+                      'data' => JSON.dump(request.message.to_h)
+                    }
+                  )
+                end
```

Comment:
Same as before: we shouldn't be dumping the whole request into the attributes. There are defined conventions that tell us what to do instead.